### DISCUSSION TITLE
Trouble running tests using Parcel

### DISCUSSION THREAD

**Discussion by chadams** [avajs/ava#3018] [Q&A]
I followed the basic setup, and my repo is uncomplicated. However my tests show
```
  Uncaught exception in tests/parser.spec.js

  SyntaxError: Cannot use import statement outside a module

  Uncaught exception in tests/test.spec.js

  SyntaxError: Cannot use import statement outside a module

  ✖ tests/parser.spec.js exited with a non-zero exit code: 1
  ✖ tests/test.spec.js exited with a non-zero exit code: 1
  ```

**Comment by novemberborn** [comment]
Going by the error message, your file contains ESM syntax however Node.js / AVA is treating it as a CommonJS file. For `*.js` files AVA will always follow the Node.js behavior, see https://nodejs.org/api/esm.html.

**Reply by chadams** [reply]
here is my repo https://github.com/chadams/js-pipes

I'm trying to understand, the AVA docs here https://github.com/avajs/ava say to make a test.js file with ESM syntax?

```
import test from 'ava';

test('foo', t => {
	t.pass();
});

test('bar', async t => {
	const bar = Promise.resolve('bar');
	t.is(await bar, 'bar');
});
```

**Reply by novemberborn** [reply]
ESM is the way the ecosystem is moving, hence us using that in the documentation. But ultimately AVA is a Node.js runner so you need to use the syntax that Node.js is expecting, or configure Node.js to work with your syntax.

**Comment by chadams** [comment]
may package.json has `"type": "module"` I don't see any other requirements, I'm on node 16+

**Reply by novemberborn** [reply]
It really depends on the specifics of the setup. Unfortunately I don't have the time to dig into yours.
